modules/mgclient.mjs

```diff
@@ -0,0 +1,1799 @@
+export {
+    default as factory
+}
+from "../examples/out.js"
+
+/**
+ *  Simple class that handles WASM resources. Every Mg datatype is allocated internally 
+ *  by the C runtime. Therefore, these resources must be deallocated at the end of execution.
+ *  Although this simple construct is used as a global resource manager it can be easily extended 
+ *  by adding scoped resource capabilities. Howerver, unless proven in practice that this is needed,
+ *  it suffices to rely on this simple utility.
+ */
+class MgResourceManagerInternal {
+    #resourceManager;
+    constructor() {
+        this.#resourceManager = [];
+    }
+
+    push(resource) {
+        this.#resourceManager.push(resource);
+    }
+
+    stopManaging(resource) {
+        let index = this.#resourceManager.indexOf(resource);
+        if (index == -1) {
+            throw "Internal error";
+        }
+        this.#resourceManager.splice(index, 1);
+    }
+
+    releaseAll() {
+        // deallocate in reverse order
```

Comment:
Why the reverse order is important (just expand the comment to be more descriptive).

Reply:
Just following the usual C++ conversion that deallocations happen in reverse order of the allocations. Nothing more to expand here ?

Reply:
This was more about improving this comment to be in the proper format because `deallocate in reverse order` is obvious from the code (the comment as-is is totally redundant). If you have any comments at all, then focus on the "because" part because this tells to the reader what was the intention / what are the implications (that part is not obvious, sometimes even in trivial cases). But all good 😄